lib/phraseapp_updater/locale_file/loader.rb

```diff
@@ -0,0 +1,24 @@
+require 'phraseapp_updater/locale_file'
+
+class PhraseAppUpdater
+  class LocaleFile
+    class Loader
+      def initialize(extension)
+        @extension = extension
+      end
+
+      def load(filename)
+        unless File.readable?(filename) && File.file?(filename)
+          raise RuntimeError.new("Couldn't read localization file at #{filename}")
+        end
+
+        LocaleFile.class_for_file_format(@extension).new(File.basename(filename).chomp(".#{@extension}"), File.read(filename))
```

Comment:
For this particular program it's not a huge deal, but technically you have a race condition here in between checking the file's readability and actually reading it. You might consider just skipping the check and attempting to read, raising that `RuntimeError` if you get an `IOError` or `SystemCallError`.